client/core/core.go

```diff
@@ -2802,6 +2802,38 @@ func (c *Core) discoverAccount(dc *dexConnection, crypter encrypt.Crypter) (bool
 	return true, nil // great, just stay connected
 }
 
+// dexWithPubKeyExists checks whether or not there is a non-disabled account
+// for a dex that has pubKey.
+func (c *Core) dexWithPubKeyExists(pubKey *secp256k1.PublicKey) (bool, string, error) {
+	c.connMtx.RLock()
+	defer c.connMtx.RUnlock()
+
+	for host, dexConnection := range c.conns {
+		if dexConnection.acct.dexPubKey == nil {
+			continue
+		}
+
+		if dexConnection.acct.dexPubKey.IsEqual(pubKey) {
+			return true, host, nil
+		}
+	}
+
+	return false, "", nil
+}
+
+// upgradeConnection promotes a temporary dex connection and starts listening
+// to the messages it recieves.
+func (c *Core) upgradeConnection(dc *dexConnection) {
+	atomic.StoreUint32(&dc.reportingConnects, 1)
+	c.wg.Add(1)
+	go c.listen(dc)
+	go dc.subPriceFeed()
```

Comment:
If you switch to the following, I believe you can use `upgradeConnection` from `(*Core).register` as well:

```go
if atomic.CompareAndSwapUint32(&dc.reportingConnects, 0, 1) {
	c.wg.Add(1)
	go c.listen(dc)
	go dc.subPriceFeed()
}
c.connMtx.Lock()
c.conns[dc.acct.host] = dc
c.connMtx.Unlock()
```